### DISCUSSION TITLE
Lesson 22 comments

### DISCUSSION THREAD

**Discussion by jumarko** [jacekschae/learn-pedestal-course-files#16] [Feedback]
Hi and thanks for a really nice episode!

I've got a few comments: 
1. I think there was a nice opportunity to use `cond->` when listing recipes - like this: 

```
(cond-> {:public public-recipes}
      account-id (merge
                  {:drafts (d/q '[:find (pull ?e pattern)
...
                   })
      true response)
...
```

2. 404 and "this is one of the gotchas with pedestal" - this is not really specific to Pedestal. If you return `nil` from your handler, Ring returns 404 - the same "problem" with Compojure, etc.

3. The workflow isn't as nice as I hoped. I have constant problems reloading the definition of the `list-recipes-response` function. I need to reload a bunch of namespaces and then restart the server to pick up the new definitions. What I'm used to is simply reloading the function and letting the system pick up the new definition automatically.
I tried to pass a var instead but it's still not working - not sure why: 
```
(def table-routes
  (route/expand-routes
...
     ["/recipes" :get #'recipes/list-recipes-response :route-name :list-recipes]
...
```

4. I had to generate the response as JSON to make it properly return the data (when trying the request in the browser)

```
(-> db-data json/generate-string response)
```
```

**Comment by jacekschae** [comment]
Thanks for your comments. Nice use of `cond->`. 

The setup is not finished yet. I'll add few things to the workflow. After setting up so many things in the first videos I wanted to start working on something that is not setup.
